src/BeadingStrategy/DistributedBeadingStrategy.cpp

```diff
@@ -1,46 +1,101 @@
-//Copyright (c) 2020 Ultimaker B.V.
-//CuraEngine is released under the terms of the AGPLv3 or higher.
-
+// Copyright (c) 2021 Ultimaker B.V.
+// CuraEngine is released under the terms of the AGPLv3 or higher.
+#include <numeric>
 #include "DistributedBeadingStrategy.h"
 
 namespace cura
 {
 
-    DistributedBeadingStrategy::Beading DistributedBeadingStrategy::compute(coord_t thickness, coord_t bead_count) const
+DistributedBeadingStrategy::DistributedBeadingStrategy(const coord_t optimal_width,
+                                const coord_t default_transition_length,
+                                const AngleRadians transitioning_angle,
+                                const Ratio wall_transition_threshold,
+                                const float distribution_radius)
+    : BeadingStrategy(optimal_width, default_transition_length, transitioning_angle),
+      wall_transition_threshold(wall_transition_threshold),
+      one_over_distribution_radius_squared(1.0f / distribution_radius * 1.0f / distribution_radius)
```

Comment:
I know that the `distribution_radius` (aka `wall_distribution_count`) can get a minimum value of 1 in the interface, but still I think there should be a check for 0 here.

Reply:
https://github.com/Ultimaker/CuraEngine/pull/1477/commits/431ccdedded35a471a83e2839ba0fb19b2c50107